src/hotspot/share/runtime/arguments.cpp

```diff
@@ -2906,6 +2906,18 @@ jint Arguments::parse_each_vm_init_arg(const JavaVMInitArgs* args, bool* patch_m
       jio_fprintf(defaultStream::error_stream(),
                   "ExtendedDTraceProbes flag is not applicable for this configuration\n");
       return JNI_EINVAL;
+    } else if (match_option(option, "-XX:+DTraceMethodProbes")) {
+      jio_fprintf(defaultStream::error_stream(),
+                  "DTraceMethodProbes flag is not applicable for this configuration\n");
```

Comment:
Looking at this again, in light of the test changes, I have to wonder if the pre-existing treatment of `ExtendedDTraceProbes` was wrong and we should simply have used `UNSUPPORTED_OPTION` in these cases?

Reply:
Maybe, but this block also handles ergonomic opt-in to `DTrace*` flags when `ExtendedDTraceProbe` is enabled, and it matches the style of the similar checks. I'd leave it as is.

Reply:
ok